graphium/trainer/predictor.py

```diff
@@ -310,13 +310,17 @@ def _general_step(self, batch: Dict[str, Tensor], step_name: str, to_cpu: bool)
         }
         # preds = {k: preds[ii] for ii, k in enumerate(targets_dict.keys())}
         for task, pred in preds.items():
+            if pred.dtype == torch.float16:
+                pred_fp32 = pred.to(torch.float32)
+                pred = pred_fp32
+                preds[task] = pred_fp32
             task_specific_norm = self.task_norms[task] if self.task_norms is not None else None
             if step_name != "train":
                 # apply denormalization for val and test predictions for correct loss and metrics evaluation
                 # targets for val and test were not normalized
                 # train loss will stay as the normalized version
                 preds[task] = task_specific_norm.denormalize(pred)
-            targets_dict[task] = targets_dict[task].to(dtype=pred.dtype)
+                targets_dict[task] = targets_dict[task].to(dtype=pred.dtype)
```

Comment:
Not sure it makes sense. Why not casting the dtype also at training?? It will crash if the targets and preds are not the same dtype.

Above, you convert to float32, but what if the targets are float16? or float64?